File: src/main/kotlin/messy_filtering/solution/java_dsl/entities/Order.java
package messy_filtering.solution.java_dsl.entities;

import jakarta.persistence.*;

@Entity
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String status;
    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    // Getters and setters
    public Long getId() {
        return id;
    }

    public String getStatus() {
        return status;
    }

    public User getUser() {
        return user;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: src/main/kotlin/messy_filtering/solution/kotlin_dsl/domain/Order.kt
package messy_filtering.solution.kotlin_dsl.domain

import jakarta.persistence.*

@Entity
@Table(name = "orders")
data class Order(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,
    val status: String,
    @ManyToOne
    @JoinColumn(name = "user_id")
    val user: User? = null,
)
